evalml/pipelines/components/estimators/clusterers/__init__.py

```diff
@@ -0,0 +1,5 @@
+"""EvalML clustering components."""
```

Comment:
Thoughts on renaming this module to `unsupervised`? Easier to say than `clusterer` imo.

Reply:
I think it's important to keep the component names as clusterers, since there are more types of unsupervised learning than just clustering. I'm happy to rename the higher-level module to unsupervised though!